android/src/main/kotlin/com/mr/flutter/plugin/filepicker/FileUtils.kt

```diff
@@ -0,0 +1,615 @@
+            TAG,
+            "Allowed file extensions mimes: $mimes"
+        )
+        return mimes
+    }
+
+    @JvmStatic
+    fun getFileName(uri: Uri, context: Context): String? {
+        var result: String? = null
+
+        try {
+            if (uri.scheme == "content") {
+                context.contentResolver.query(
+                    uri,
+                    arrayOf(OpenableColumns.DISPLAY_NAME),
+                    null,
+                    null,
+                    null
+                ).use { cursor ->
+                    if (cursor != null && cursor.moveToFirst()) {
+                        result =
+                            cursor.getString(cursor.getColumnIndexOrThrow(OpenableColumns.DISPLAY_NAME))
+                    }
+                }
+            }
+            if (result == null) {
+                result = uri.path?.substringAfterLast('/')
+            }
+        } catch (ex: Exception) {
+            Log.e(
+                TAG,
+                "Failed to handle file name: $ex"
+            )
+        }
+
+        return result
+    }
+
+    @JvmStatic
+    fun isImage(context: Context, uri: Uri): Boolean {
+        val extension = getFileExtension(context, uri)
+        return (extension != null && (extension.contentEquals("jpg") || extension.contentEquals("jpeg") || extension.contentEquals(
+            "png"
+        ) || extension.contentEquals("WEBP")))
+    }
+
+    private fun getFileExtension(context: Context, uri: Uri): String? {
+        val contentResolver = context.contentResolver
+        val mimeType = contentResolver.getType(uri)
+        return MimeTypeMap.getSingleton().getExtensionFromMimeType(mimeType)
+    }
+
+    private fun getCompressFormat(context: Context, uri: Uri): Bitmap.CompressFormat {
+        val format = getFileExtension(context, uri)
+        return when (format!!.uppercase(Locale.getDefault())) {
+            "PNG" -> Bitmap.CompressFormat.PNG
+            "WEBP" -> Bitmap.CompressFormat.WEBP
+            else -> Bitmap.CompressFormat.JPEG
+        }
+    }
+
+    @JvmStatic
+    fun compressImage(originalImageUri: Uri, compressionQuality: Int, context: Context): Uri {
+        val compressedUri: Uri
+        try {
+            context.contentResolver.openInputStream(originalImageUri).use { imageStream ->
+                val compressedFile = createImageFile(context, originalImageUri)
+                val originalBitmap = BitmapFactory.decodeStream(imageStream)
+                // Compress and save the image
+                val fos = FileOutputStream(compressedFile)
+                originalBitmap.compress(
+                    getCompressFormat(context, originalImageUri),
+                    compressionQuality,
+                    fos
+                )
+                fos.flush()
+                fos.close()
+                compressedUri = Uri.fromFile(compressedFile)
+            }
+        } catch (e: IOException) {
+            throw RuntimeException(e)
```

Comment:
Can we rethrow the IOException? I'd rather not get reports from runtime exceptions :p